### PR TITLE
HTMLFormatter - Improve inline and eex tags break

### LINKED ISSUE #2001: Heex formatter changes HTML
### Environment

* Elixir version (elixir -v): Elixir 1.13.2 (compiled with Erlang/OTP 24)
* Phoenix version (mix deps): 1.6.7
* Phoenix LiveView version (mix deps): 0.17.9
* Operating system: MacOS
* Browsers you attempted to reproduce this bug on (the more the merrier): n/a (issue with rendered HTML rather than browser behaviour)
* Does the problem persist after removing "assets/node_modules" and trying again? Yes

### Actual behavior

This Heex template:

```
<p>Read <%= @my_managers_name %>'s feedback. Reading their feedback in advance will help you to get the most value from your Progress Review meeting!</p>
```

Got reformatted to:

```
  <p>
    Read <%= @my_managers_name %>
    's feedback. Reading their feedback in advance will help you to get the most value from your Progress Review meeting!
  </p>
```

Which is bad as when it renders there is now a space between `my_managers_name` and the apostrophe.

### Expected behavior

Don't insert newlines where there wasn't a space before

### PR DESCRIPTION
**Pull request by leandrocp**
That's an attempt to improve and close #2001 
Ref https://github.com/phoenixframework/phoenix_live_view/issues/2001#issuecomment-1113645932

I had to fine tune the inline break rules a bit to deal with all possible cases, and we can add more tests if you have other edge cases in mind.

/cc @feliperenan thanks for commenting about that `flex_break`

### CODE REVIEW

File: test/phoenix_live_view/html_formatter_test.exs [hunk 1]
```diff
@@ -87,6 +87,45 @@ if Version.match?(System.version(), ">= 1.13.0") do
       assert_formatter_doesnt_change(input)
     end
 
+    test "do not break between eex tags when there is no space before or after" do
```
Comment (feliperenan) [inline test/phoenix_live_view/html_formatter_test.exs:90]: We need to the same for inline tags. I guess the current code doesn't cover:

```elixir
      assert_formatter_output(
        """
        <p>first <span>name</span>second</p>
        """,
        """
        <p>
          first <span>name</span>second
        </p>
        """,
        line_length: 20
      )
```

Comment (leandrocp) [reply test/phoenix_live_view/html_formatter_test.exs:90]: > We need to the same for inline tags

You're right It doesn't cover inline tags, this PR is focused on eex tags only. I guess it's fine to add it in this PR, I'm gonna work on it.

File: lib/phoenix_live_view/html_algebra.ex [hunk 2]
```diff
@@ -97,6 +117,12 @@ defmodule Phoenix.LiveView.HTMLAlgebra do
 
   defp text_ends_with_space?(_node), do: false
 
+  defp is_text_node?({:text, _text, _meta}), do: true
+  defp is_text_node?(_node), do: false
+
+  defp is_eex_node?({:eex, _expr, _meta}), do: true
```
[the remark this answers is not in the document]
Comment (leandrocp) [reply lib/phoenix_live_view/html_algebra.ex]: @feliperenan just pushed some changes https://github.com/phoenixframework/phoenix_live_view/pull/2006/commits/6d85bb4c6d588cdfb451214bdf3da198e6051eeb

Is that what you had in mind?

File: test/phoenix_live_view/html_formatter_test.exs [hunk 3]
```diff
@@ -1276,18 +1313,17 @@ if Version.match?(System.version(), ">= 1.13.0") do
...
         <p>
-          text text <a class="text-blue-500" href="" target="_blank" attr1="">link</a>text.
+          first first <a class="text-blue-500" href="" target="_blank" attr1="">link</a>second.
         </p>
         """,
         """
         <p>
-          text text
-          <a
+          first first <a
```
Comment (leandrocp) [inline test/phoenix_live_view/html_formatter_test.exs:1321]: This may seem wrong but the inline element `a` is followed by a text `second` without whitespace causing it to sticky on the same line and breaking its attrs into new lines because of the lack of space in the same line caused by  `line_length: 50`

File: lib/phoenix_live_view/html_algebra.ex (hunk 2 above)
[the remark this answers is not in the document]
Comment (leandrocp) [reply lib/phoenix_live_view/html_algebra.ex]: In your diff, the assertion https://github.com/phoenixframework/phoenix_live_view/commit/1d9525a78bddfd9807309edd2923099642433e66#diff-71b84bb0b0ae5fed38f8063817963199429180245db2a517c01805466372b003L468 shouldn't be left with no changes? 🤔 
I mean the eex tag is surrounded by space and the last group of text is beyond the limit, which should cause a line break right?

[the remark this answers is not in the document]
Comment (leandrocp) [reply lib/phoenix_live_view/html_algebra.ex]: > Would you mind checking if that breaks some rule in your codebase?

Checking it out now and it's looking good! I still see a few issues with line reset but it's out of scope, we can deal with it in another PR.
